project/api/models.py

```diff
@@ -87,9 +90,17 @@ def clean(self, *args, **kwargs):
         if self.start_date > self.end_date:
             raise ValidationError("Start date cannot be later than end date")
         if loads(self.geo.json)["type"] != "Polygon" and loads(self.geo.json)["type"] != "MultiPolygon":
-            raise ValidationError("Only Polygon and MultiPolygon objects are acceptable geometry types")
+            raise ValidationError(
+                "Only Polygon and MultiPolygon objects are acceptable geometry types")
         super(Territory, self).clean(*args, **kwargs)
 
+        print "asd"
+        territories = Territory.objects.filter(nation=self.nation)
+        for territory in territories:
+            if (territory.start_date <= self.start_date <= territory.end_date) or (territory.start_date <= self.end_date <= territory.end_date):
+                raise ValidationError(
```

Comment:
Incorrect: if there's one existing territory with date [3;4], and you create a new one with date [1;5], then this check not not throw an error.

Reply:
Should we fine now if I didn't forget another case?